src/cert_context.rs

```diff
@@ -471,17 +478,23 @@ impl<'a> AcquirePrivateKeyOptions<'a> {
             }
             assert!(free == winapi::TRUE);
             if spec & wincrypt::CERT_NCRYPT_KEY_SPEC != 0 {
-                Ok(PrivateKey::NcryptKey(NcryptKey::from_inner(handle)))
-            } else {
-                Ok(PrivateKey::CryptProv(CryptProv::from_inner(handle)))
-            }
+                return Ok(PrivateKey::NcryptKey(NcryptKey::from_inner(handle)));
+            } 
+            
+            #[cfg(feature = "allow-deprecated")] 
+            return Ok(PrivateKey::CryptProv(CryptProv::from_inner(handle)));
+            
+            #[cfg(not(feature = "allow-deprecated"))] 
+            return Err(io::Error::new(io::ErrorKind::Other, "Api failure"));
+            
         }
     }
 }
 
 /// The private key associated with a certificate context.
 pub enum PrivateKey {
     /// A CryptoAPI provider.
+    #[cfg(feature = "allow-deprecated")]
```

Comment:
Doing it this way leads to a breaking change. 
I'm fine with changing the default to CRYPT_ACQUIRE_PREFER_NCRYPT_KEY_FLAG, as that should not break anything.
We can mark it `#[deprecated]` but I don't think there's any advantage to hiding it behind a feature?